table/block_based/full_filter_block.cc

```diff
@@ -101,19 +104,33 @@ void FullFilterBlockBuilder::Reset() {
   last_prefix_recorded_ = false;
 }
 
-Slice FullFilterBlockBuilder::Finish(
-    const BlockHandle& /*tmp*/, Status* status,
-    std::unique_ptr<const char[]>* filter_data) {
+Status FullFilterBlockBuilder::Finish(const BlockHandle& /*prev*/,
+                                      MemoryAllocator* allocator,
+                                      CacheAllocationPtr* output_buf,
+                                      Slice* output_filter) {
   Reset();
-  // In this impl we ignore BlockHandle
-  *status = Status::OK();
   if (any_added_) {
     any_added_ = false;
-    Slice filter_content = filter_bits_builder_->Finish(
-        filter_data ? filter_data : &filter_data_, status);
-    return filter_content;
+    Status s = filter_bits_builder_->FinishV2(allocator, output_filter);
+    if (output_filter->size() > 0) {
+      assert(s.ok());
```

Comment:
Suppose `output_filter` is pointed to some existing filter already (which is not the current codebase but [FilterBlockBuilder::Finish()](https://github.com/facebook/rocksdb/pull/9581/files?show-viewed-files=true&file-filters%5B%5D=#diff-3e272453769b8a4018265dada5511164458bb741a2d63bc633ed0eb75ff7d531R80-R86)'s API doc does not warn me from doing so) and suppose corruption in construction has left `output_filter` unmodified, seems to me that this assertion will break.

I am thinking maybe we could do:
```
if (s.ok()){
  if (output_filter->size() > 0) {
   if (output_buf) {
    ...
   }
  } else {
      // Should not generate empty filter if any added
      assert(!s.ok());
  }
} else {
   // do what we need to do for !s.ok()
}
```